Content.Server/GameObjects/Components/Mobs/Speech/SpanishAccentComponent.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.CodeAnalysis.Operations;
+using Microsoft.EntityFrameworkCore.Internal;
+using Robust.Shared.GameObjects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Content.Server.GameObjects.Components.Mobs.Speech
+{
+    [RegisterComponent]
+    public class SpanishAccentComponent : Component, IAccentComponent
+    {
+        public override string Name => "SpanishAccent";
+
+        public string Accentuate(string message)
+        {
+            // Insert E before every S
+            message = InsertS(message);
+            // Insert
+            message = ReplaceQuestionMark(message);
+            return message;
+        }
+
+        private static string InsertS(string message)
```

Comment:
really wondering why this isn't just:

```cs
public static string spanish(string message){
	var ret = message.Replace(" s", " es").Replace(" S", "Es");
	if(ret.StartsWith("s")){
		return ret.Remove(0, 1).Insert(0, "es");
	} else if(ret.StartsWith("S")){
		return ret.Remove(0, 1).Insert(0, "Es");
	}
	return ret;
}
```

Same for your other hand-rolled string manipulations, I don't think performance is a major concern here, so the maintainability gains of using simple stdlib methods is good.

Reply:
Because they were basically written by 2 different persons (me and me from a week ago)